docs/proposals/project-scoped-appset.md

```diff
@@ -0,0 +1,114 @@
+
+### Goals
+
+Add support for deploying ApplicationSets inside a AppProject and limit the interaction with elements inside it. 
+This includes also the RBAC verification
+
+### Non-Goals
+
+This doesn't include any change in the UI/cli to make them user friendly
+
+## Proposal
+
+The proposal for achieving this goal is to add a "required" `project` field (read more below) into the ApplicationSet and based on it, limit the access to the resources during the Applications generation. 
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-addons
+  namespace: argocd
+spec:
+  project: sample-project # new field
+  generators:
+  - ...
+```
+
+As we want to maintain the option of deploying global ApplicationSet (without any restriction), administrators will be able to deploy appsets without setting the project field. For achieving this, we will check if the user has access to **ALL** `applicationsets` across **ALL** the projects (*/*) and only that case, they will be able to deploy `applicationsets` without setting the project:
+
+```
+p, role:admin, applicationsets, create, */*, allow
+p, role:admin, applicationsets, update, */*, allow
+```
+
+Other option is to be explicit and create an specific permission for this:
+
+```
+p, role:admin, applicationsets, unscoped, *, allow
+```
+
+> Note: `unscoped` is a proposal, but maybe `global` or any other name is better.
+
+Inside the appset controller, the controller will check if appset has set the `project` field and it can produce 2 different behaviors:
+
+1. `project` isn't set: In this case, nothing will change from current behavior
+2. `project` is set: In this case, **ALL** the resources involved in the Application generation have to be scoped to the project where the appset is deployed. 
+In case of cluster/git generators the cluster/repos have to be project scoped, but in other generators, the controller will check if the resource is allowed for the project (for example, using a list generator the controller will block forbidden clusters/repos)
+
+### Use cases
+
+#### Use case 1:
+In my company, the platform team manages shared services like ArgoCD and we have a shared instance for all the projects. Currently we cannot use ArgoCD with this approach because the team A could deploy things (voluntarily or not) into other teams clusters. We could use Applications instead of ApplicationSet but in that case, we will lose the option of spinning up a production cluster ready to work in a few minutes (kubernetes cluster deployment time + apps startup).
```

Comment:
I feel like this use case might already be solved by the `spec.template.spec.project` field - if that field is hard-coded with some project's name, then users are allowed to create ApplicationSets, using that project, via the API.

Reply:
Yes, this use case is totally solved by `spec.template.spec.project` 🙇